app/scripts/models/reliers/relier.js

```diff
@@ -44,11 +52,18 @@ define([
      *         // do overriding behavior here.
      *       });
      * }
+     *
+     * @method fetch
      */
     fetch: function () {
       var self = this;
       return p()
         .then(function () {
+          // parse the resume token before importing any other data.
+          // query parameters and server provided data override
+          // resume provided data.
+          self.populateFromStringifiedResumeToken(self.getSearchParam('resume'));
```

Comment:
We had a [test already](https://github.com/mozilla/fxa-content-server/pull/2688/files#diff-271e3329d7651c7cb1cef5747adcd1dfR38), I updated the name of the test to reflect the goal.